cmd/starter/c/starter.c

```diff
@@ -286,6 +292,52 @@ static int apply_container_privileges(struct privileges *privileges) {
 #endif
 }
 
+static void set_rpc_privileges(void) {
+    struct privileges *priv = (struct privileges *)malloc(sizeof(struct privileges));
+
+    if ( priv == NULL ) {
+        fatalf("could not allocate memory\n");
+    }
+
+    memset(priv, 0, sizeof(struct privileges));
+
+    priv->capabilities.effective |= capflag(CAP_MKNOD);
+    priv->capabilities.effective |= capflag(CAP_SETGID);
+    priv->capabilities.effective |= capflag(CAP_SETUID);
+    priv->capabilities.effective |= capflag(CAP_SYS_ADMIN);
+    priv->capabilities.effective |= capflag(CAP_SYS_CHROOT);
+
+    priv->capabilities.permitted = priv->capabilities.effective;
+    priv->capabilities.bounding = priv->capabilities.effective;
+
+    debugf("Set RPC privileges\n");
+    apply_privileges(priv);
+
+    free(priv);
+}
+
+static void set_master_privileges(void) {
+    struct privileges *priv = (struct privileges *)malloc(sizeof(struct privileges));
+
+    if ( priv == NULL ) {
+        fatalf("could not allocate memory\n");
+    }
+
+    memset(priv, 0, sizeof(struct privileges));
+
+    priv->capabilities.effective |= capflag(CAP_SETGID);
+    priv->capabilities.effective |= capflag(CAP_SETUID);
+
+    priv->capabilities.permitted = 0xffffffffff;
```

Comment:
What's the a reason we are masking like this rather than setting equal to effective?

Is there a smaller set of capabilities we would need later on?

Reply:
We may need arbitrary capabilities (CNI), so we just keep setuid/setgid as effective and the rest will be raised automatically during an execve as root user by example, if we set all capabilities as effective spawned process couldn't raise those capabilities